### PR TITLE
Connect Zone HVAC Unit Ventilator to DOA

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
This feature allows connecting Unit Ventilator zone HVAC equipment to central dedicated air (DOA) to provide outdoor (conditioned or unconditioned) air. The DOA can be connected to the inlet or outlet side of the Unit Ventilator using the  AirTerminal:SingleDuct:Mixer object.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [x] Functional code review (it has to work!)
 - n/a If defect, results of running current develop vs this branch should exhibit the fix
 - [x] CI status: all green or justified
 - [x] Performance: CI Linux results include performance check -- verify this
 - [x] Unit Test(s)
 - C++ checks:
   - [x] Argument types
   - [x] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [x] Verify naming conventions and styles, memos and notes and defaults
   - [x] Open windows IDF Editor with modified IDD to check for errors
   - n/a If transition, add rules to spreadsheet
 - [x] If new idf included, locally check the err file and other outputs
 - [x] Documentation changes in place
 - [x] Changed docs build successfully
 - n/a ExpandObjects changes?
 - n/a If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: testfiles/CMakeLists.txt [hunk 1]
```diff
@@ -173,6 +173,7 @@ ADD_SIMULATION_TEST(IDF_FILE DOAToFanCoilSupply.idf EPW_FILE USA_IL_Chicago-OHar
 ADD_SIMULATION_TEST(IDF_FILE DOAToPTAC.idf EPW_FILE USA_FL_Miami.Intl.AP.722020_TMY3.epw)
 ADD_SIMULATION_TEST(IDF_FILE DOAToPTHP.idf EPW_FILE USA_FL_Miami.Intl.AP.722020_TMY3.epw)
 ADD_SIMULATION_TEST(IDF_FILE DOAToUnitarySystem.idf EPW_FILE USA_FL_Miami.Intl.AP.722020_TMY3.epw)
+ADD_SIMULATION_TEST(IDF_FILE DOAToUnitVentilator.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
```
Comment (mjwitte) [inline testfiles/CMakeLists.txt:177]: This file did not get checked in, so this is failing [here](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-5df6da2034ca72c768570b6a0b245515384f5ac5-x86_64-Linux-Ubuntu-14.04-gcc-4.8-IntegrationCoverage-Debug.html).
Also, this PR needs a description.

File: idd/Energy+.idd.in [hunk 2]
```diff
@@ -33627,15 +33627,18 @@ ZoneHVAC:UnitVentilator,
         \required-field
         \type node
    A8,  \field Outdoor Air Node Name
-        \required-field
         \type node
+        \note this field is left blank only if the Unit Ventilator is connected to a central
```
Comment (mjwitte) [inline idd/Energy+.idd.in]: ZoneHVAC:UnitVentilator, Name field needs
\reference DOAToZonalUnit

File: tst/EnergyPlus/unit/AirTerminalSingleDuctMixer.unit.cc [hunk 3]
```diff
@@ -6716,4 +6718,481 @@ namespace EnergyPlus {
 		{ IOFlags flags; flags.DISPOSE( "DELETE" ); gio::close( OutputFileInits, flags ); }
 	}
 
+		TEST_F( EnergyPlusFixture, AirTerminalSingleDuctMixer_SimUnitVent_ATMInletSide ) {
+
+		int write_stat;
+		OutputFileInits = GetNewUnitNumber();
+		{ IOFlags flags; flags.ACTION( "write" ); flags.STATUS( "UNKNOWN" ); gio::open( OutputFileInits, "eplusout.eio", flags ); write_stat = flags.ios(); }
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/AirTerminalSingleDuctMixer.unit.cc]: These three lines to open the eio file (and the close later on) are not needed.  You can remove this here and in all of the other tests in this file.

File: tst/EnergyPlus/unit/AirTerminalSingleDuctMixer.unit.cc [hunk 4]
```diff
@@ -6716,4 +6718,481 @@ namespace EnergyPlus {
...
+		SimUnitVentilator( UnitVent( UnitVentNum ).Name, ZoneNum, FirstHVACIteration, QUnitOut, LatOutputProvided, UnitVentNum );
+		// apply mass conservation to determine secondary air mass flow rate
+		SecondaryAirMassFlowRate = Node( UnitVent( UnitVentNum ).AirInNode ).MassFlowRate - PrimaryAirMassFlowRate;
+		// check the air mixer secondary air mass flow rate 
+		ASSERT_EQ( SecondaryAirMassFlowRate, Node( SysATMixer( 1 ).SecInNode ).MassFlowRate );
+		// check the cooling output delivered is within 2.0 Watt of zone cooling load 
+		ASSERT_NEAR( QZnReq, QUnitOut, 0.001 );
+
+		// Close and delete eio output file
+		{ IOFlags flags; flags.DISPOSE( "DELETE" ); gio::close( OutputFileInits, flags ); }
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/AirTerminalSingleDuctMixer.unit.cc]: Delete here and elsewhere in this file.

File: idd/Energy+.idd.in (hunk 2 above)
Comment (mjwitte) [reply idd/Energy+.idd.in]: Inlet Node Name field needs notes explaining that it must be a zone exhaust node if there is no DOA mixer or if there is an outlet side DOA mixer, but the unit ventilator air inlet node name must be the same as the mixer outlet node name for an inlet side mixer.  I/O ref description for the Inlet Node field needs the same notes.

File: doc/input-output-reference/src/overview/group-zone-forced-air-units.tex [hunk 5]
```diff
@@ -891,9 +891,9 @@ \subsubsection{Outputs}\label{outputs-1-033}
...
... and exhaust nodes and the outdoor air by specifying unit inlet, outlet, outdoor air and exhaust (relief) air node names. Note that if the Unit Ventilator is not connected to central dedicated outdoor air (DOA), then the unit air inlet node should be the same as a zone exhaust air node and the unit air outlet node should be the same as a zone air inlet node. The three node name input fields: \textit{Outdoor Air Node Name}, \textit{Exhaust Air Node Name}, and \textit{Mixed Air Node Name} are required fields only when the built-in outdoor air mixer is used to provide outdoor air. If the Unit Ventilator is connected to a central dedicated outdoor air (DOA) via \textit{AirTerminal:SingleDuct:Mixer}, then the above three node name input fields are not required and must be left blank. In the later case the Unit Ventilator node connection to the zone air exhaust node and the zone air inlet node are described in \textit{AirTerminal:SingleDuct:Mixer} object. In general, the unit ventilator input is very similar to the fan coil unit input, and the unit is connected to a hot water loop (demand side) through its hot water coil and to a chilled water loop (demand side) through its cooling coil.
```
Comment (mjwitte) [inline doc/input-output-reference/src/overview/group-zone-forced-air-units.tex:897]: Engineering Reference Unit Ventilator section has the same (or almost the same) paragraph, so it needs the same changes.

**Review by mjwitte — CHANGES_REQUESTED** [review]
@Nigusse Some initial review comments.  I have not tested the code yet - need the example file which didn't get checked in.

File: doc/input-output-reference/src/overview/group-zone-forced-air-units.tex (hunk 5 above)
Comment (Nigusse) [reply doc/input-output-reference/src/overview/group-zone-forced-air-units.tex:897]: @mjwitte I have addressed your review comments. I am testing if the documents build locally. I will push the changes shortly.

**Review by mjwitte — APPROVED** [review]

**Comment by mjwitte** [comment]
@Nigusse Everything looks good except there are over 3000 `SimHVAC: Exceeding Maximum iterations` warnings when running the new test file as an annual simulation.  I ran a couple of other DOA test files (PTAC and FanCoil) as annuals, and they do not have these warnings.  Please take a look.

**Comment by Nigusse** [comment]
I am looking at it.

**Comment by Nigusse** [comment]
@mjwitte Revised the example file to eliminate the SimHVAC max iteration error warnings. The Unit Vent availability and outdoor air min schedules were revised. Also did cleanup.  I will commit the revised example shortly.

**Comment by mjwitte** [comment]
@Nigusse Much better, thanks.  This will go in after CI reports come back.

**Comment by mjwitte** [comment]
@Myoldmopar Missed the warnings over in #5929 that _HybridVentilationControl_MinTime.idf was not in the cmake list file.  Correcting that here.

**Pull request merged**